agent/connect/csr.go

```diff
@@ -13,10 +14,21 @@ import (
 
 // CreateCSR returns a CSR to sign the given service along with the PEM-encoded
 // private key for this certificate.
-func CreateCSR(uri CertURI, privateKey crypto.Signer, extensions ...pkix.Extension) (string, error) {
+func CreateCSR(commonName string, uri CertURI, privateKey crypto.Signer,
+	extensions ...pkix.Extension) (string, error) {
+
+	signAlgo := x509.SHA256WithRSA
+	_, ok := privateKey.(*ecdsa.PrivateKey)
+	if ok {
+		signAlgo = x509.ECDSAWithSHA256
+	}
```

Comment:
🤔 is this missing a case for RSA? I'm not sure why it was changed if we still only support one value...

Reply:
No, the default is RSA (line 20). The algorithm is only changed to ECDSA if the cast succeeds.